backend/app/services/diagnosis_agent.py

```
@@ -0,0 +1,486 @@
+    if region in weather_cache:
+        cached_time, cached_data = weather_cache[region]
+        if now - cached_time < WEATHER_CACHE_TTL:
+            return {"weather_data": cached_data}
+            
+    api_key = os.getenv("WEATHER_API_KEY")
+    if not api_key:
+        return {"weather_data": {"temperature": "에러", "humidity": "-", "wind_speed": "-", "precipitation_prob": "-"}}
+
+    import requests
+    from datetime import datetime, timedelta
+    current_time = datetime.now()
```

Ruff (0.15.10)

[warning] `datetime.datetime.now()` called without a `tz` argument

(DTZ005)